### ISSUE TITLE
Java: Exception is thrown when saveInformationAsync is called on a non existent collection

### ISSUE THREAD

**Issue by johnoliver** [microsoft/semantic-kernel#1776]
```
com.microsoft.semantickernel.memory.MemoryException: Attempted to access non-existent collection: foo
        at com.microsoft.semantickernel.memory.VolatileMemoryStore.getCollection(VolatileMemoryStore.java:242)
        at com.microsoft.semantickernel.memory.VolatileMemoryStore.upsertAsync(VolatileMemoryStore.java:67)
        at
```

**Comment by d3r3kk** [comment]
To verify and fix or close as necessary.

**Comment by dsgrieve** [comment]
Fixed by PR #2306
